Assets/Scripts/SS3D/Systems/Chat/ChatFilterOption.cs

```diff
@@ -0,0 +1,22 @@
+﻿using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace SS3D.Engine.Chat
+{
+    public class ChatFilterOption : MonoBehaviour
+    {
+        [SerializeField] private TextMeshProUGUI label = null;
+        [SerializeField] private Toggle toggle = null;
+
+        private ChatChannel _channel;
+
+        public void Init(ChatChannel channel)
+        {
+            _channel = channel;
+            label.text = channel.name;
+        }
+
+        public ChatChannel TickedChannel() => toggle.isOn ? _channel : null;
```

Comment:
Returning null is bad in most cases, that can lead to unexpected behaviour, I'd rather see a TryGetTickedChannel() outing the ChatChannel instead of that.